astronomer/providers/snowflake/operators/snowflake.py

```diff
@@ -31,6 +38,33 @@
 from astronomer.providers.utils.typing_compat import Context
 
 
+def _check_queries_finish(conn: SnowflakeConnection, query_ids: list[str]) -> bool:
+    """Check whether snowflake queries finish (in aborting, failed_with_error or success)"""
+    with closing(conn) as conn:
+        for query_id in query_ids:
+            status = conn.get_query_status(query_id)
+
+            if status == QueryStatus.ABORTING:
+                raise AirflowException(f"ABORTING {ABORTING_MESSAGE}")
+            elif status == QueryStatus.FAILED_WITH_ERROR:
+                raise AirflowException(f"FAILED_WITH_ERROR {FAILED_WITH_ERROR_MESSAGE}")
+            elif status == QueryStatus.SUCCESS:
+                pass
+            elif status in (
+                QueryStatus.RUNNING,
+                QueryStatus.QUEUED,
+                QueryStatus.DISCONNECTED,
+                QueryStatus.RESUMING_WAREHOUSE,
+                QueryStatus.BLOCKED,
+                QueryStatus.NO_DATA,
+            ):
+                return False
+            else:
+                raise ValueError(f"Unexpected QueryStatus: {str(status)}")
```

Comment:
There is another status FAILED_WITH_INCIDENT, I guess we could also raise an exception for it.

Reply:
It doesn't seem to be supported by [snowflake-python-connector](https://docs.snowflake.com/developer-guide/python-connector/python-connector-api#enums)

Reply:
Looks like that document is missing it. But it does seem to have it 
<img width="1633" alt="Screenshot 2023-04-25 at 6 45 34 PM" src="https://user-images.githubusercontent.com/10206082/234288183-4bb24651-c1e5-446e-8478-d5fe3237947d.png">

Reply:
You're right. I should have looked into the source code. Doc could be misleading. Thanks! Just added it